### DISCUSSION TITLE
support for relative links in Markdown to files in submodules

### DISCUSSION THREAD

**Discussion by mfulton26** [community/community#51141] [Repositories]
### Select Topic Area

Product Feedback

### Body

I found something unexpected while using github.com to view files in repositories using git submodules: "404 - page not found":

1. given two repositories named "parent" and "child" where "child" is mapped to a folder inside "parent" as a git submodule
2. and a Markdown file in "parent" containing relative links to Markdown files in "child"
3. and I'm viewing a Markdown file in the "parent" repository via github.com
4. when I click a relative link to a Markdown file in the "child" folder
5. then I expect the Markdown file in the "child" folder to load instead of getting "404 - page not found"

e.g. if I'm viewing [https://github.com/{{ownerOf(parent)}}/parent/blob/main/README.md](https://github.com/%7B%7BownerOf(parent)%7D%7D/parent/blob/main/README.md) and I click a link I want to be able to navigate to [https://github.com/{{ownerOf(child)}}/child/blob/main/README.md](https://github.com/%7B%7BownerOf(child)%7D%7D/child/blob/main/README.md)

i.e. I want [https://github.com/{{ownerOf(parent)}}/parent/blob/main/child/README.md](https://github.com/%7B%7BownerOf(parent)%7D%7D/parent/blob/main/child/README.md) to redirect to [https://github.com/{{ownerOf(child)}}/child/blob/main/README.md](https://github.com/%7B%7BownerOf(child)%7D%7D/child/blob/main/README.md) because "child" here is a git submodule in "parent"

I could use an absolute URL but then when working in the git repository locally I'd be taken to the web browser instead of to the local file inside the git submodule

Feature request: support resolving files inside git submodules to their actual URLs on github.com for better file navigation in web browsers

**Comment by mirenradia** [comment]
I have also just discovered that GitHub does not currently support this which is a bit annoying. Note that users have wanted this feature for quite a while: see github/markup#1068.

**Reply by SusanWoodhouse** [reply]
Seriously?! So annoying!

**Reply by lvaroqui-gpfw** [reply]
That's really annoying indeed...

**Reply by brzuchal** [reply]
That'd be really nice to have!

**Reply by davidattwater** [reply]
Joining my annoyed voice to the throng ....

**Comment by davidstephensonfocusrite** [comment]
I would very much like to see a response from GitHub on this point, even if the response is simply that this is not a priority. Getting no response at all from github seems a little unreasonable.
